pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -54,12 +55,15 @@ def on_train_batch_start(
         batch_idx: int,
         unused: Optional[int] = 0,
     ) -> None:
+        if not trainer.logger:
+            raise MisconfigurationException("Cannot use `DeviceStatsMonitor` callback with `Trainer(logger=False)`.")
```

Comment:
@ananthsub why did you add two more `MisconfigurationException`s here if we already have one in line 48?

Reply:
probably because of mypy, which does not know that the logger was already checked against None in the previous hook :))